pkg/envoy/server.go

```diff
@@ -935,27 +944,39 @@ func getDirectionNetworkPolicy(l4Policy policy.L4PolicyMap, policyEnforced bool)
 						// the other rules.
 						allowAll = true
 					}
-					pnp.Rules = append(pnp.Rules, rule)
+					rules = append(rules, rule)
 				}
 			}
 		}
 		// Short-circuit rules if a rule allows all and all other rules can be short-circuited
 		if allowAll && canShortCircuit {
 			log.Debug("Short circuiting HTTP rules due to rule allowing all and no other rules needing attention")
-			pnp.Rules = nil
+			rules = nil
 		}
 
 		// No rule for this port matches any remote identity.
 		// This means that no traffic was explicitly allowed for this port.
 		// In this case, just don't generate any PortNetworkPolicy for this
 		// port.
-		if !allowAll && len(pnp.Rules) == 0 {
+		if !allowAll && len(rules) == 0 {
 			continue
 		}
 
+		// Add rules to a new or existing entry for this port
+		key := fmt.Sprintf("%d/%s", port, l4.Protocol)
```

Comment:
Might worth create a function that returns the a Key when given a port and a protocol.

Reply:
This is actually used in quite a many places:
```
$ git grep "\"%d/%s\""
cilium/cmd/bpf_policy_get.go:                   port = fmt.Sprintf("%d/%s", dport, proto.String())
pkg/envoy/server.go:            key := fmt.Sprintf("%d/%s", port, l4.Protocol)
pkg/policy/l4.go:                       port := fmt.Sprintf("%d/%s", l4Ctx.Port, lwrProtocol)
pkg/policy/policy.go:           dports = append(dports, fmt.Sprintf("%d/%s", dport.Port, dport.Protocol))
pkg/policy/visibility.go:                       pp := fmt.Sprintf("%d/%s", portInt, prot.String())
```

But I will leave this for a future enhancement for now. The map in this case is a local one, so there need not be a contract about the map key with anyone else.